modules/fedimint-mint-client/src/lib.rs

```diff
@@ -1325,13 +1326,23 @@ impl MintClientModule {
                     match state.state {
                         MintOOBStates::TimeoutRefund(refund) => Some(SpendOOBRefund {
                             user_triggered: false,
-                            transaction_id: refund.refund_txid,
+                            transaction_ids: vec![refund.refund_txid],
                         }),
                         MintOOBStates::UserRefund(refund) => Some(SpendOOBRefund {
                             user_triggered: true,
-                            transaction_id: refund.refund_txid,
+                            transaction_ids: vec![refund.refund_txid],
                         }),
-                        MintOOBStates::Created(_) => None,
+                        MintOOBStates::UserRefundMulti(_refund) => None,
+                        MintOOBStates::RefundedMulti => Some(SpendOOBRefund {
+                            user_triggered: true,
+                            // this state already waited for txid, it was a success
+                            transaction_ids: vec![],
+                        }),
+                        MintOOBStates::RefundedMultiFallback(refund) => Some(SpendOOBRefund {
+                            user_triggered: true,
+                            transaction_ids: refund.refund_txids,
+                        }),
```

Comment:
In the old state machines the OOB spend SMs weren't responsible for reclaiming e-cash in case the original refund tx got rejected. Instead the e-cash input SMs of the refund tx would take care of reclaiming these notes. I think you actually implemented this redundantly here.

If we keep the old model and only make it multi-note then the code becomes much simpler and we could likely even get rid of old states because we wouldn't need to keep notes around inside `RefundMulti`.

Reply:
> In the old state machines the OOB spend SMs weren't responsible for reclaiming e-cash in case the original refund tx got rejected. 

There was no "original refund". It would create bunch of `claim_input` and was done.

Do you mean that the failed `claim_input` should attempt splitting into claim-each-note-separately? Hmmm... That might be quite nice, and it would do it for everything ...

@elsirion

Reply:
@elsirion So before, when we did per-note `claim_input` on refunding OOB, we would do another round try to reclaim the reclaim?